### PR TITLE
feat(pre-commit): automatically fix html anchors in github wikis

### PR DESCRIPTION
**Pull request by Swiftb0y**
See @Holzhaus's suggestion. https://github.com/mixxxdj/manual/pull/481#issuecomment-1118773443

### CODE REVIEW

File: tools/fixup_gh_wiki_anchors.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+)
+URL_SUB = r"\g<url>#user-content-\g<anchor>"
+
+
+def main(argv=None):
+    parser = argparse.ArgumentParser()
+    parser.add_argument("input_file", nargs="+")
+    args = parser.parse_args(argv)
+    for filepath in args.input_file:
+        file_content = ""
```
Comment (Holzhaus) [inline tools/fixup_gh_wiki_anchors.py:17]: Not needed. 

```suggestion
```

File: tools/fixup_gh_wiki_anchors.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        with open(filepath, "r") as file:
+            file_content = file.read()
+        with open(filepath, "w") as file:
+            subbed = re.sub(URL_REGEX, URL_SUB, file_content)
```
Comment (Holzhaus) [inline tools/fixup_gh_wiki_anchors.py:21]: If we swap these, we can reduce the duration the file is open. 

```suggestion
        subbed = re.sub(URL_REGEX, URL_SUB, file_content)
        with open(filepath, "w") as file:
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -10,6 +10,11 @@ repos:
     language: python
     additional_dependencies:
     - scour==0.38.2
+  - id: gh-wiki-anchors
+    name: "Fix GitHub Wiki Anchors for Linkcheck"
+    files: ^source/.*\.rst$
```
Comment (Holzhaus) [inline .pre-commit-config.yaml:15]:
```suggestion
    types: [rst]
```

**Review by Holzhaus — COMMENTED** [review]
Thanks, code looks good. Didn't test yet.

**Comment by Swiftb0y** [comment]
Thanks, Done.

**Comment by daschuer** [comment]
Thank you.

**Pull request merged**